src/Datadog.Trace.ClrProfiler.Managed/NativeMethods.cs

```diff
@@ -1,10 +1,36 @@
+using System;
 using System.Runtime.InteropServices;
 
+// ReSharper disable MemberHidesStaticFromOuterClass
 namespace Datadog.Trace.ClrProfiler
 {
     internal static class NativeMethods
     {
-        [DllImport("Datadog.Trace.ClrProfiler.Native")]
-        public static extern bool IsProfilerAttached();
+        public static bool IsProfilerAttached()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
What about OSPlatform.OSX?

Reply:
Not supported for auto-instrumentation.

Reply:
Alternatively, we could use `.dll` on Windows and drop it for any other platform, since .NET Core doesn't require the extension.

Reply:
Oh wait, we don't produce OSX assets do we?

Reply:
We don't, but if someone gets the C++ code to build on OSX it would be nice if this worked.

Reply:
Changed the logic in b678451f75c83775efb756d6a5a2e6c3fff7d84b.